ks_patches/innreise_duplicates.js

```diff
@@ -0,0 +1,141 @@
+
+    selectedTei.enrollments.forEach(enrollment => {
+        if (enrollment.program === DUPLICAT_INNREISE_PROGRAM_CODE) {
+            enrollment.events.forEach(event => {
+                try {
+                    var newEvent = angular.copy(event);
+                    newEvent.program = INNREISE_PROGRAM_CODE;
+                    newEvent.programStage = duplicateToInnreiseEventCode(event.programStage);
+                    var id = Array.isArray(enrollmentId) ? enrollmentId[0] : enrollmentId;  // Needed to avoid "Event date is required."-error
+                    newEvent.enrollment = id;
+                    if (newEvent.notes) {
+                        newEvent.notes = newEvent.notes.map(note => {
+                            delete note.note; // Cannot reuse note ID
+                            return note;
+                        });
+                    }
+                    delete newEvent.event; // Cannot reuse event ID
+                    if (newEvent.programStage) {
+                        eventFactory.create(newEvent);
+                    }
+                } catch (err) {
+                    console.log(err);
+                    throw err;
+                }
+            });
+        }
+    });
+}
+
+function updateProfile(selectedTei, existingTei, optionSets, attributesById, teiService) {
+    selectedTei.attributes.forEach((attribute) => {
+        if (!existingTei.attributes.some((att) => att.code === attribute.code)) {
+            existingTei.attributes.push(attribute);
+        }
+    });
+
+    return teiService.update(existingTei, optionSets, attributesById);
+}
+
+function createProfile(selectedTei, optionSets, attributesById, teiService) {
+    var apiTei = cleanTeiForCreate(selectedTei);
+    return teiService.register(apiTei, optionSets, attributesById).then((response) => {
+        return response.response.importSummaries[0].reference;
+    });
+}
+
+function createEnrollment(exisitngTeiId, orgUnitId) {
+    var enrollment = {};
+    enrollment.trackedEntityInstance = exisitngTeiId;
+    enrollment.program = INNREISE_PROGRAM_CODE;
+    enrollment.status = 'ACTIVE';
+    enrollment.orgUnit = orgUnitId;
+    return enrollment;
+}
+
+function cleanTeiForCreate(tei) {
+    delete tei.enrollment;
+    delete tei.programOwnersById;
+    delete tei.relationships;
+    delete tei.trackedEntityInstance;
+    delete tei.trackedEntityType;
+    tei.trackedEntityType = INNREISE_ENTITY_TYPE;
+    return tei;
+}
+
+
+function duplicateToInnreiseEventCode(duplicateCode) {
+    if (duplicateCode === DUPLIKAT_OPPFOLGING_STAGE_CODE) {
+        return OPPFOLGING_STAGE_CODE;
+    }
+    if (duplicateCode === DUPLIKAT_INNREISE_STAGE_CODE) {
+        return INNREISEINFORMASJON_PROGRAM_STAGE_CODE;
+    }
+    return undefined;
+}
+
+function isEnrolledInInnreise(tei) {
+    return tei.enrollments.some(enrollment => enrollment.program === INNREISE_PROGRAM_CODE);
```

Comment:
You should also check wether enrollment is open

Reply:
AKA ` enrollment.status === 'ACTIVE'`?